package/MDAnalysis/core/Selection.py

```diff
@@ -624,90 +602,115 @@ class BaseSelection(NucleicSelection):
      'N9', 'N7', 'C8', 'C5', 'C4', 'N3', 'C2', 'N1', 'C6',
      'O6','N2','N6', 'O2','N4','O4','C5M'
     """
-    base_atoms = {x: None for x in [
+    base_atoms = [
         'N9', 'N7', 'C8', 'C5', 'C4', 'N3', 'C2', 'N1', 'C6',
         'O6', 'N2', 'N6',
-        'O2', 'N4', 'O4', 'C5M']}
+        'O2', 'N4', 'O4', 'C5M']
 
-    def _apply(self, group):
-        return set([a for a in group.atoms
-                    if (a.name in self.base_atoms and
-                        a.resname in self.nucl_res)])
+    def apply(self, group):
+        mask = np.in1d(group.names, self.base_atoms)
+        mask &= np.in1d(group.resnames, self.nucl_res)
+        return unique(group[mask])
 
 
 class NucleicSugarSelection(NucleicSelection):
     """Contains all atoms with name C1', C2', C3', C4', O2', O4', O3'.
     """
-    sug_atoms = {x: None for x in ['C1\'', 'C2\'', 'C3\'', 'C4\'', 'O4\'']}
+    sug_atoms = ["C1'", "C2'", "C3'", "C4'", "O4'"]
 
-    def _apply(self, group):
-        return set([a for a in group.atoms
-                    if (a.name in self.sug_atoms and
-                        a.resname in self.nucl_res)])
+    def apply(self, group):
+        mask = np.in1d(group.names, self.sug_atoms)
+        mask &= np.in1d(group.resnames, self.nucl_res)
+        return unique(group[mask])
 
 
-class PropertySelection(Selection):
+class PropertySelection(object):
     """Some of the possible properties:
     x, y, z, radius, mass,
     """
-    def __init__(self, prop, operator, value, abs=False):
+    def __init__(self, prop, operator, value, abs_=False):
         self.prop = prop
         self.operator = operator
         self.value = value
-        self.abs = abs
-
-    def _apply(self, group):
-        # For efficiency, get a reference to the actual np position arrays
-        if self.prop in ("x", "y", "z"):
-            p = getattr(Selection.coord, '_' + self.prop)
-            indices = np.array([a.index for a in group.atoms])
-            if not self.abs:
-                # XXX Hack for difference in np.nonzero between version < 1. and version > 1
-                res = np.nonzero(self.operator(p[indices], self.value))
-            else:
-                res = np.nonzero(self.operator(np.abs(p[indices]), self.value))
-            if type(res) == tuple:
-                res = res[0]
-            result_set = [group.atoms[i] for i in res]
-        elif self.prop == "mass":
-            result_set = [a for a in group.atoms
-                          if self.operator(a.mass, self.value)]
-        elif self.prop == "charge":
-            result_set = [a for a in group.atoms
-                          if self.operator(a.charge, self.value)]
-        return set(result_set)
-
-
-class SameSelection(Selection):
+        self.abs_ = abs_
```

Comment:
I would leave it as `self.abs`, that's what the name space of a class does :-).
I dislike the trailing underscore for aesthetic reasons, so I'd rather write it out as `absolute`.